plugins/services/src/js/containers/service-connection/SDKServiceConnectionContainer.js

```diff
@@ -0,0 +1,270 @@
+
+const METHODS_TO_BIND = [
+  "handleOpenEditConfigurationModal",
+  "handleActionDisabledModalClose",
+  "handleTextCopy"
+];
+
+class SDKServiceConnectionContainer extends React.Component {
+  constructor() {
+    super(...arguments);
+
+    this.state = {
+      actionDisabledModalOpen: false,
+      copiedCommand: false
+    };
+
+    METHODS_TO_BIND.forEach(method => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  handleTextCopy(copiedCommand) {
+    this.setState({ copiedCommand });
+  }
+
+  handleOpenEditConfigurationModal() {
+    this.setState({
+      actionDisabledModalOpen: true
+    });
+  }
+
+  handleActionDisabledModalClose() {
+    this.setState({ actionDisabledModalOpen: false });
+  }
+
+  componentDidMount() {
+    SDKEndpointActions.fetchEndpoints(this.props.service.getId());
+  }
+
+  getClipboardTrigger(command) {
+    return (
+      <div className="code-copy-wrapper">
+        <div className="code-copy-icon tight">
+          <ClipboardTrigger
+            className="clickable"
+            copyText={command}
+            onTextCopy={this.handleTextCopy.bind(this, command)}
+            useTooltip={true}
+          >
+            <Icon id="clipboard" size="mini" ref="copyButton" color="grey" />
+          </ClipboardTrigger>
+        </div>
+        {command}
+      </div>
+    );
+  }
+
+  getJSONEndpoint(endpoint) {
+    return (
+      <ConfigurationMapSection>
+        <ConfigurationMapHeading>
+          {endpoint.getEndpointName()}
+        </ConfigurationMapHeading>
+        <ConfigurationMapRow>
+          <ConfigurationMapLabel>
+            Address
+          </ConfigurationMapLabel>
+          <ConfigurationMapValue>
+            <span>{this.getClipboardTrigger(endpoint.getAddress())}</span>
+          </ConfigurationMapValue>
+        </ConfigurationMapRow>
+        <ConfigurationMapRow>
+          <ConfigurationMapLabel>
+            DNS
+          </ConfigurationMapLabel>
+          <ConfigurationMapValue>
+            {this.getClipboardTrigger(endpoint.getDns())}
+          </ConfigurationMapValue>
+        </ConfigurationMapRow>
+        <ConfigurationMapRow>
+          <ConfigurationMapLabel>
+            VIP
+          </ConfigurationMapLabel>
+          <ConfigurationMapValue>
+            {this.getClipboardTrigger(endpoint.getVip())}
+          </ConfigurationMapValue>
+        </ConfigurationMapRow>
+      </ConfigurationMapSection>
+    );
+  }
+
+  getFileEndpoint(endpoint) {
+    return (
+      <ConfigurationMapRow>
+        <ConfigurationMapLabel>
+          {endpoint.getEndpointName()}
+        </ConfigurationMapLabel>
+        <ConfigurationMapValue>
+          <a
+            className="active endpoint-download"
+            download={endpoint.getEndpointName()}
+            href={`data:text/plain;content-disposition=attachment;filename=${endpoint.getEndpointName()};charset=utf-8,${encodeURIComponent(endpoint.getEndpointData())}`}
```

Comment:
Not sure if it makes sense to extract this into a Util or a copmonent

Reply:
Created a new Util for this